llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -676,7 +676,9 @@ PassBuilder::buildFunctionSimplificationPipeline(OptimizationLevel Level,
 
   // Try vectorization/scalarization transforms that are both improvements
   // themselves and can allow further folds with GVN and InstCombine.
-  FPM.addPass(VectorCombinePass(/*TryEarlyFoldsOnly=*/true));
+  // Disable for SYCL due to performance issues.
+  if (!SYCLOptimizationMode)
```

Comment:
The regression we have is actually ESIMD only, so in theory I could try to modify the VectorCombine pass itself to skip ESIMD functions, but it won't be that simple because:
1) The regression in question is in user code called from ESIMD, so it does not have the ESIMD function attributes
2) Even if we try to solve 1) by traversing up the callgraph, we won't be able to detect functions called indirectly.